fix: client connected InvokeOnClientConnectedCallback with scene management disabled

NoelStephensUnity opened the pull request:
This fixes the issue where a client would not receive the OnClientConnectedCallback event when scene management was disabled.
It also does some slight refactoring to when it is invoked as well as invokes it on the server after the client has notified the server that its synchronization is complete (as opposed to invoking right when the client was approved) in order to assure a user can start sending RPCs to a client from a server when this is invoked.  
Side note: Users can also use the C2S_SyncComplete scene event notification to determine if it is ok to send RPCs to a client.

This includes an OnClientConnectedCallback integration test to verify this is working when scene management is enabled and disabled.

_This includes a bump in the number of NetworkBufferPools to remove the spamming which is causing performance issues and finally reached a point where it was causing MessageHandlerReceivedMessageServerClient to fail due to the warning messages._

NoelStephensUnity committed:
fix

Fix for the client connected message not being invoked on the client if scene management is disabled.
This includes an integration test, MultiClientConnectionApproval.ClientConnectedCallbackTest,  to verify client connected is being invoked when scene management is both enabled and disabled.

NoelStephensUnity committed:
fix

This is a fix to reduce the amount of console spew due to network buffer pools exceeding 1024.  This reduces unit test processing time and console log sizes, while also providing a more realistic buffer pool size.

NoelStephensUnity committed:
test

The unit test.

NoelStephensUnity committed:
refactor

Removing this minor LF as the file has nothing to do with this PR.

NoelStephensUnity committed:
style

removing 2 LF that were not intended

NoelStephensUnity committed:
refactor style

Some straggler style requests from PR-1080 that are harmless and piggy-backing onto this PR.

JesseOlmer commented on com.unity.netcode.gameobjects/Runtime/Serialization/Pooled/NetworkBufferPool.cs:
```diff
@@ -12,8 +12,11 @@ public static class NetworkBufferPool
+        // Increasing this value for various reasons:
+        // Some tests (i.e. MessageHandlerReceivedMessageServerClient) expect certain log messages and if we happen to hit the threshold it will cause those tests to fail
+        // This slows down tests having to log warnings repeatedly.
+        private const uint k_MaxBitPoolBuffers = 2048;
```
This feels like it's just hiding a problem in our tests, or worse, in the framework... Why are we using more than a thousand buffers simultaneously?

0xFA11 replied:
+1 :(

NoelStephensUnity replied:
Ok, I can put it back to 1024. Our unit tests will consume much more log file space and MessageHandlerReceivedMessageServerClient will fail because it is detecting the warning that we are exceeding 1024 buffers.

It is a terrible pooling system, I was just trying to provide plenty of room.  It isn't the amount we are allocating, it is the amount required before it spams a warning message.

NoelStephensUnity replied:
Just reverted the changes, this PR won't be able to pass Yamato tests.

NoelStephensUnity replied:
Well that's just crazy...it passed even though it fails on my end.
Go figure.,

0xFA11 replied:
there is an old JIRA ticket here: MTT-656

NoelStephensUnity replied:
Yeah, increasing the k_MaxBitPoolBuffers only really prevents the warning message spam during unit tests which reduces processing overhead, time, and the size of the log files.  It doesn't pre-allocate buffers based on that size...just doesn't start warning until it reaches that threshold.   Keeping it at 1024 just means lots of console spam during unit and integration tests.

NoelStephensUnity committed:
refactor

Reverting the NetworkBufferPool changes.
This will make MessageHandlerReceivedMessageServerClient fail because of the warning message that we have exceeded 1024 buffers.

NoelStephensUnity committed:
Merge branch 'develop' into fix/clientconnected-no-scene-management
